python/cuml/tests/test_simpl_set.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from cuml.datasets import make_blobs
+import numpy as np
+import cupy as cp
+import umap.distances as dist
+from cuml.manifold.umap import UMAP
+from umap.umap_ import fuzzy_simplicial_set as ref_fuzzy_simplicial_set
+from cuml.manifold.simpl_set import fuzzy_simplicial_set \
+    as cu_fuzzy_simplicial_set
+from umap.umap_ import simplicial_set_embedding as ref_simplicial_set_embedding
+from cuml.manifold.simpl_set import simplicial_set_embedding \
+    as cu_simplicial_set_embedding
+
+
+def correctness_dense(a, b, rtol=0.1, threshold=0.8):
+    n_elms = a.size
+    n_correct = (cp.abs(a - b) <= (rtol * cp.abs(b))).sum()
+    correctness = n_correct / n_elms
+    return correctness >= threshold
+
+
+def correctness_sparse(a, b, atol=0.1, rtol=0.2, threshold=0.8):
+    n_ref_zeros = (a == 0).sum()
+    n_ref_non_zero_elms = a.size - n_ref_zeros
+    n_correct = (cp.abs(a - b) <= (atol + rtol * cp.abs(b))).sum()
+    correctness = (n_correct - n_ref_zeros) / n_ref_non_zero_elms
+    return correctness >= threshold
+
+
+def test_fuzzy_simplicial_set():
```

Comment:
Can we parametrize this for a couple different input sizes/n_neighbors combinations?